### DISCUSSION TITLE
Debian 12 and Budgie Desktop

### DISCUSSION THREAD

**Discussion by makitso** [BuddiesOfBudgie/budgie-desktop#413] [Q&A] [answered]
Wanted to try the budgie desktop on debian 12 (gnome based).  Install completed and logged to my account.  However, there is no way to manage desktop icons, folders, files, etc.  Can't drag trash can. 
My question is, what is the best base desktop to install debian with and then do the install of budgie?

**Comment by fossfreedom** [comment]
On debian the default desktop icons support is provided by budgie-desktop-view

You can substitute bdv with nemo or desktop-folder for alternative desktop icons.

You will need to tweak a budgie key for desktop icons via budgie desktop settings ... something like

https://github.com/UbuntuBudgie/budgie-desktop-environment/blob/5e4b5a11ef20518c54909851fa5098fad48442a3/schemas/25_budgie-desktop-environment.gschema.override#L149

**Comment by makitso** [comment]
Thanks David.  I can not drag any desktop icons, trash, etc, to anyplace on the desktop, almost like the keep aligned option was set.
I can right click and see the Budgie Desktop setting option.  I am using standard nautilus for desktop mgt.

**Comment by fossfreedom** [comment]
The fixed positioning of BDV is by design.

The drag and drop is not something I am aware of since I don't use nautilus
but I suspect it is something todo with gtk4 apps like nautilus dropping
onto a gtk3 based app like BDV

**Reply by makitso** [reply]
So, I recreated Debian 12 using the mate desktop.  With that desktop I can move icons around.
I installed Budgie Desktop and View. After reboot I could not move icons but could add them via budgie menu.
Installed Nemo and gave it control of desktop.  Still have no ability to move trash can from left side of screen to right.

**Reply by EbonJaeger** [reply]
I'm not sure what steps you've done or what "gave it control of desktop" means, but I would recommend going through the steps I wrote in [this post](https://discuss.getsol.us/d/9115-switching-from-budgie-desktop-view-to-nemo-desktop). It was written with Solus in mind, but it should work for any distro; just ignore the bit about `budgie-desktop-branding`.

@fossfreedom would have to say whether `budgie-desktop-view` can be uninstalled on Debian or not; I don't know if it's set as a dependency of `budgie-desktop` or a metapackage or whatever. If not, then somehow `budgie-desktop-view` will have to be kept from automatically starting.

**Reply by makitso** [reply]
Followed your procedure but still did not work.

>  Run this in a terminal (or set it via Dconf Editor) to tell Budgie to use Nemo for the desktop view: gsettings set com.solus-project.budgie-wm desktop-type-override nemo.

Since am running in Mate OS, is this commend correct?

>Make sure you have the package budgie-desktop-branding-fortitude-plus installed as well. It contains a few settings overrides to make Nemo Desktop work with conky, Showtime, and desktop_weather.

Can't find this package.

**Reply by fossfreedom** [reply]
> Since am running in Mate OS, is this commend correct?

Obviously not - best ask on a more appropriate forum

> Can't find this package.

Well EbonJaeger did say to ignore that part of the instructions.

**Comment by fossfreedom** [comment]
BDV is a recommendation. So it can be removed without uninstalling the budgie-desktop

**Comment by makitso** [comment]
Thanks David.

I really like the budgie desktop and run ubuntu budgie on all of my laptops. 
Debian 12 is released with a lot of desktops but not budgie.  I looked on the internet and it appeared that installing the budgie desktop on Debian was relatively easy task –  but not for me!  

The first issue was  deciding what base desktop do you install Debian with?  I tried gnome and mate but came up with the same issue;  could get budgie to control the desktop icons but could not move them around on the desktop e.g.  move the trash can from top left to nay other spot.  Right clicking on the desktop did not allow for creation of a folder or file.

Now, I am just playing around with this.  But, IMHO, it seems to me that installing budgie on Debian should not take this much effort.  Jist my 2 cents.

**Reply by fossfreedom** [reply]
All true.

In the near future I will be uploading a "budgie-desktop-environment" package that will configure things automatically like desktop icons, filemanager defaults, theme etc.

I will then do a PR on tasksel so that when installing from the [netinst ISO](https://www.debian.org/CD/netinst/) you can pick budgie from the list of available desktops.

In the mean time - I personally test 'debian testing' via installing netinst without any desktop.  At the command prompt on reboot install slick-greeter, gnome-terminal, budgie-desktop, arc-theme, papirus-icon-theme, nemo, firefox-esr - followed by removing budgie-desktop-view to give you a minimal desktop that you can easily then set the theme and icons to via budgie-desktop-settings.

I follow this up by manually tweaking etc/apt/sources.list to switch to unstable + adding experimental - living on the edge!

**Comment by makitso** [comment, marked as answer]
Again, thanks David.
